projects/kubernetes/python/README.md

```diff
@@ -0,0 +1,198 @@
+Root Dir: /var/lib/docker/aufs
+Dirs: 289
+Execution Driver: native-0.2
+Kernel Version: 3.16.0-4-amd64
+Operating System: Debian GNU/Linux 8 (jessie)
+WARNING: No memory limit support
+WARNING: No swap limit support
+```
+
+### kubectl is a command-line interface for executing commands against a Kubernetes cluster. Run the shell script below to install kubectl
+
+```
+curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
+
+```
+### Deploying to Kubernetes requires a containerized application. Let's review containerizing Python applications.
+
+### Containerization at a glance
+### Containerization involves enclosing an application in a container with its own operating system. This full machine virtualization option has the advantage of being able to run an application on any machine without concerns about dependencies.
+
+### Create a Python container image
+
+## To create these images, we will use Docker, which enables us to deploy applications inside isolated Linux software containers. Docker is able to automatically build images using instructions from a Docker file.
+
+## This is a Docker file for our Python application:
+
+````
+
+FROM python:3.6
+MAINTAINER XenonStack
+# Creating Application Source Code Directory
+RUN mkdir -p /k8s_python_sample_code/src
+# Setting Home Directory for containers
+WORKDIR /k8s_python_sample_code/src
+# Installing python dependencies
+COPY requirements.txt /k8s_python_sample_code/src
+RUN pip install --no-cache-dir -r requirements.txt
+# Copying src code to Container
+COPY . /k8s_python_sample_code/src/app
+# Application Environment variables
+ENV APP_ENV development
+# Exposing Ports
+EXPOSE 5035
+# Setting Persistent data
+VOLUME ["/app-data"]
+# Running Python Application
+CMD ["python", "app.py"]
+
+``````
+
+## Build a Python Docker image
+
+##Build the Docker image using this command:
+##
+##docker build -t k8s_python_sample_code
+##
+##Publish the container images
+##We can publish our Python container image to different private/public cloud repositories, like Docker Hub, AWS ECR, Google Container Registry, etc. For this tutorial, we'll use Docker Hub.
+##
+##Tag a version to the image, before building the image
+##
+```
+docker tag k8s_python_sample_code:latest k8s_python_sample_code:0.1
+```
+##
+##Push the image to a cloud repository
+##
+##Using a Docker registry other than Docker Hub to store images requires to add that container registry to the local Docker daemon and Kubernetes Docker daemons. Docker Hub is used in this example.
+##
+##Execute this Docker command to push the image:
+
+```
+docker push k8s_python_sample_code
+```
+
+##Working with CephFS persistent storage
+##
+##Kubernetes supports many persistent storage providers, including AWS EBS, CephFS, GlusterFS, Azure Disk, NFS, etc.
+##To use CephFS for persistent data to Kubernetes containers, we will create two files:
```

Comment:
Working with CephFS persistent storage

Kubernetes supports many persistent storage providers, including AWS EBS, CephFS, GlusterFS, Azure Disk, NFS, etc.
To use CephFS for persistent data to Kubernetes containers, we will create two files: